File: lesson_19/src/main/java/org/lesson19/Lot.java
package org.lesson19;

import jakarta.persistence.*;

@Entity
@Table(name = "lot")
public class Lot {
    @Id
    @Column(name = "id")
    private Long id;

    @Column(name = "title")
    private String title;

    @Column(name = "price")
    private Integer price;

    public Lot(Long id, String title) {
        this.id = id;
        this.title = title;
        this.price = 0;
    }

    @Version
    private Long version;

    public Lot() {
    }

    public Long getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Integer getPrice() {
        return price;
    }

    public void setPrice(Integer price) {
        this.price = price;
    }

    public User getUser() {
        return user;
    }

    @Override
    public String toString() {
        return "Lot{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", price=" + price +
                ", user=" + user +
                '}';
    }

    @OneToOne
    @JoinColumn(name = "user_id")
    User user;

    public void setUser(User user) {
        this.user = user;
    }
    // ссылку на владельца последней ставки.
}

File: lesson_19/src/main/kotlin/org/lesson19kt/Lot.kt
package org.lesson19kt

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.OneToOne
import jakarta.persistence.Table
import jakarta.persistence.Version

@Entity
@Table(name = "lot")
data class Lot(
    @Id
    @Column(name = "id")
    var id: Long? = null,
    @Column(name = "title")
    var title: String? = null,
    @Column(name = "price")
    var price: Long? = null,
    @Version
    var version: Long? = null,

    @OneToOne
    @JoinColumn(name = "user_id")
    var user: User? = null
)